src/clouds/owncloud.js

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (C) 2018 Orange.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Rombit - initial API and implementation
+ ******************************************************************************/
+
+// const debug = require('debug-levels')('agile-data');
+// const agile = require('./agile-sdk');
+// const get = require('lodash/get');
+const path = require('path');
+const request = require('request');
+
+module.exports = {
+  endpointDescription: function () {
+    return({
+      "cloud": "owncloud",
+    })
+  },
+  upload: function(credentials, remotepath,payload) {
+    return new Promise(function(resolve, reject) {
+
+      if (typeof remotepath !== 'undefined' && typeof remotepath !== "undefined"  && typeof payload !== "undefined") {
+      let uploadConfig = {
+        uploadContent: JSON.stringify(payload),
+        uploadFileName: (typeof remotepath !== 'undefined') ? remotepath : dateFormat(new Date(), "yyyy.mm.dd") + "-Agile.txt",
```

Comment:
This check is not needed, `remotepath` can not be undefined, because then the check on line 29 would fail.

Reply:
Also fixed, please let me know if I am missing something here, and the change was not needed.